src/main/java/gregtech/common/items/behaviors/ItemMagnetBehavior.java

```diff
@@ -0,0 +1,132 @@
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+public class ItemMagnetBehavior implements IItemBehaviour {
+
+    private final int range;
+    private final float speed;
+    private boolean isActive;
+
+    public ItemMagnetBehavior(int range, float speed) {
+        this.range = range;
+        this.speed = speed;
+        MinecraftForge.EVENT_BUS.register(this);
+    }
+
+    @Override
+    public ActionResult<ItemStack> onItemRightClick(World world, @Nonnull EntityPlayer player, EnumHand hand) {
+        if (!player.world.isRemote) {
+            this.isActive = !this.isActive;
+            player.sendMessage(new TextComponentTranslation(isActive ? "behavior.item_magnet.enabled" : "behavior.item_magnet.disabled"));
+        }
+        return ActionResult.newResult(EnumActionResult.PASS, player.getHeldItem(hand));
+    }
+
+    @Override
+    public void onUpdate(ItemStack itemStack, Entity entity) {
+        if (!isActive || !(entity instanceof EntityPlayer) || entity.getEntityWorld().isRemote)
+            return;
+
+        EntityPlayer entityPlayer = (EntityPlayer) entity;
+        List<EntityItem> itemsInRange = entityPlayer.getEntityWorld().getEntitiesWithinAABB(EntityItem.class, getAreaBoundingBox(entityPlayer));
+        for (EntityItem entityItem : itemsInRange) {
+            if (entityItem.isDead) continue;
+
+            double distanceX = (entityPlayer.getPosition().getX() + 0.5) - entityItem.posX;
+            double distanceY = (entityPlayer.getPosition().getY() + 0.5) - entityItem.posY;
+            double distanceZ = (entityPlayer.getPosition().getZ() + 0.5) - entityItem.posZ;
+            double distance = MathHelper.sqrt(distanceX * distanceX + distanceZ * distanceZ);
+            if (distance >= 0.7) {
+                if (!entityItem.cannotPickup()) {
+                    if (!drainEnergy(true, itemStack, (int) distance))
+                        return;
+
+                    drainEnergy(false, itemStack, (int) distance);
+
+                    double directionX = distanceX / distance;
+                    double directionY = distanceY / distance;
+                    double directionZ = distanceZ / distance;
+                    entityItem.motionX = directionX * speed * 8;
+                    entityItem.motionY = directionY * speed * 8;
+                    entityItem.motionZ = directionZ * speed * 8;
+                    entityItem.velocityChanged = true;
+                    entityItem.setPickupDelay(1);
+                }
+            } else if (!entityItem.cannotPickup()) {
+                ItemStack stack = entityItem.getItem();
+
+                ItemStack remainder = ItemHandlerHelper.insertItemStacked(new ItemStackHandler(entityPlayer.inventory.mainInventory), stack, false);
+                if (remainder.isEmpty()) {
+                    entityItem.setDead();
+                } else if (stack.getCount() > remainder.getCount()) {
+                    entityItem.setItem(remainder);
```

Comment:
If we don't insert all items into the player's inventory, do we want to refresh the lifespan of the item?